src/hooks/useTransactions.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { ethers } from "ethers";
+import { useQueries, useQueryClient, UseQueryResult } from "react-query";
+
+type Transaction = ethers.providers.TransactionResponse;
+type TransactionReceipt = ethers.providers.TransactionReceipt;
+
+type TransactionState = {
+  transactions: Transaction[];
+  transactionQueries?: UseQueryResult<TransactionReceipt>[];
+  error?: Error;
+};
+
+type TransactionsManagerState = {
+  addTransaction: (
+    tx: Transaction,
+    onComplete?: (tx: TransactionReceipt) => void
+  ) => void;
+} & TransactionState;
+function useTransactionsManager() {
+  const [transactions, setTransactions] = React.useState<Transaction[]>([]);
+  const [error, setError] = React.useState<Error>();
```

Comment:
Should this be split into two files? Divide the context and the hook separately?

Reply:
I think it could be a good point to split it into two files but I'd say let's keep this consistent with the other hooks that use context in the repo